src/identity-providers/add/AddMapper.tsx

```diff
@@ -532,16 +531,17 @@ export const AddMapper = () => {
           </Button>
           <Button
             variant="link"
-            onClick={() =>
-              history.push(
-                toIdentityProvider({
+            component={(props) => (
+              <Link
+                {...props}
+                to={toIdentityProvider({
                   realm,
-                  providerId: providerId,
+                  providerId,
```

Comment:
Once `providerId` is required you can use the non-null assertion operator here.
```suggestion
                  providerId: providerId!,
```